src/org/pentaho/reporting/platform/plugin/async/PentahoAsyncReportExecution.java

```diff
@@ -18,13 +19,13 @@
 
 public class PentahoAsyncReportExecution implements IAsyncReportExecution<InputStream> {
 
-  private final SimpleReportingComponent reportComponent;
-  private final AsyncJobFileStagingHandler handler;
-  private final String url;
-  private final String auditId;
-  private final IPentahoSession safeSession;
+  protected final SimpleReportingComponent reportComponent;
+  protected final AsyncJobFileStagingHandler handler;
+  protected final String url;
+  protected final String auditId;
+  protected final IPentahoSession safeSession;
 
```

Comment:
There is a need to set these variables as protected since they are going to be re-used in PIR plugin. As far as making listener public. Since PIR don't follow reporting package structure, but has it as dependency.